src/api/app/controllers/webui/request_controller.rb

```diff
@@ -95,9 +93,18 @@ def modify_review
   def show
     @diff_limit = params[:full_diff] ? 0 : nil
     @diff_to_superseded_id = params[:diff_to_superseded]
-    @is_author = @bs_request.creator == User.possibly_nobody.login
+    @actions = @bs_request.webui_actions(filelimit: @diff_limit, tarlimit: @diff_limit, diff_to_superseded: @diff_to_superseded, diffs: false)
 
+    @is_author = @bs_request.creator == User.possibly_nobody.login
     @is_target_maintainer = @bs_request.is_target_maintainer?(User.session)
+
+    reviews = @bs_request.reviews.where(state: 'new')
+    @my_open_reviews = reviews.select { |review| review.matches_user?(User.session) }
+    @can_add_reviews = @bs_request.state.in?([:new, :review]) && (@is_author || @is_target_maintainer || @my_open_reviews.present?)
+
+    # TODO: Remove till the end of the method once request_show_redesign is rolled out
+    return if Flipper.enabled?(:request_show_redesign, User.session)
```

Comment:
I suggest using a good old `if/else`. Under the `if` statement, we do everything we need for the beta version and in the `else` statement, everything which is for the current version. This will create duplication, but it's only temporary. Once we are done with the redesign, the migration will be super simple, we simply remove the `else` part of the code and the duplication is gone.

Reply:
I applied this suggestion.

I used `Flipper.enabled?(:request_show_redesign, User.session)` instead of `feature_enabled?(:request_show_redesign)` because the last one renders a 404 in case that the feature is not enabled.

@hennevogel, @dmarcoux please take a look again.